src/net/http.rs

```diff
@@ -241,3 +359,73 @@ pub(crate) async fn post_form<T: Serialize + ?Sized>(
     let bytes = response.collect().await?.to_bytes();
     Ok(bytes)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::time::Duration;
+
+    use crate::test_utils::TestContext;
+    use crate::tools::SystemTime;
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_http_cache() -> Result<()> {
+        let t = &TestContext::new().await;
+
+        assert_eq!(http_cache_get(t, "https://webxdc.org/").await?, None);
+
+        let html_response = Response {
+            blob: b"<!DOCTYPE html> ...".to_vec(),
+            mimetype: Some("text/html".to_string()),
+            encoding: None,
+        };
+
+        let xdc_response = Response {
+            blob: b"PK...".to_vec(),
+            mimetype: Some("application/octet-stream".to_string()),
+            encoding: None,
+        };
+        let xdc_editor_url = "https://apps.testrun.org/webxdc-editor-v3.2.0.xdc";
+        let xdc_pixel_url = "https://apps.testrun.org/webxdc-pixel-v2.xdc";
+
+        http_cache_put(t, "https://webxdc.org/", &html_response).await?;
+
+        assert_eq!(http_cache_get(t, xdc_editor_url).await?, None);
+        assert_eq!(http_cache_get(t, xdc_pixel_url).await?, None);
+        assert_eq!(
+            http_cache_get(t, "https://webxdc.org/").await?,
+            Some(html_response.clone())
+        );
+
+        http_cache_put(t, xdc_editor_url, &xdc_response).await?;
+        assert_eq!(
+            http_cache_get(t, xdc_editor_url).await?,
+            Some(xdc_response.clone())
+        );
+
+        http_cache_cleanup(t).await?;
+        assert_eq!(
+            http_cache_get(t, "https://webxdc.org/").await?,
+            Some(html_response.clone())
+        );
+
+        // HTML expires after 1 hour, but .xdc does not.
+        SystemTime::shift(Duration::from_secs(3600 + 100));
+        assert_eq!(http_cache_get(t, "https://webxdc.org/").await?, None);
+        assert_eq!(
+            http_cache_get(t, xdc_editor_url).await?,
+            Some(xdc_response.clone())
+        );
+
+        // 35 days later pixel .xdc expires because we did not request it for 35 days and 1 hour.
+        // But editor is still there because we did not request it for just 35 days.
+        SystemTime::shift(Duration::from_secs(3600 * 24 * 35 - 100));
+        assert_eq!(
+            http_cache_get(t, xdc_editor_url).await?,
+            Some(xdc_response.clone())
+        );
+        assert_eq!(http_cache_get(t, xdc_pixel_url).await?, None);
+
+        Ok(())
```

Comment:
this test does not seem to touch the code path where housekeeping removed the blob file while the html-cache entry is still there.

Reply:
Now this is checked by checking that `http_cache_get()` returns None several lines above